src/UniqueKey/Service.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+namespace SilverStripe\UniqueKey;
```

Comment:
As much as a like your presentation about feature based namespace last week, I'm a bit reluctant to adopt the idea right away in framework.

If we see this has having a direct tie in with DataObject, maybe put it in the under ORM. If we see this has something a bit more generic that could be reuse somewhere else, but it under Core.